lib/interceptor.js

```diff
@@ -47,6 +51,13 @@ function Interceptor(scope, uri, method, requestBody, interceptorOptions) {
     this.optional = false;
 }
 
+Interceptor.prototype.needsLeadingSlash = function needsLeadingSlash(uri) {
+   return uri.length > 0            // uri is not empty
+          && uri.indexOf            // uri is a String object
+          && uri.indexOf('*') !== 0 // uri doesn't start with a wildcard
+          && uri.indexOf('/') !== 0 // uri doesn't already have a leading slash
```

Comment:
After you rebase, you should get a note in CI about whether code coverage changed. I think the way this is written – a series of conditionals – it will be hard to cover all of it. Do we ever pass a non-string, for example?

I wonder if this could be simplified and inlined in the constructor where it's used:

```js
!uri.startsWith('*') && !uri.startsWith('/')
```

Reply:
Sure, I'll rebase and move the code into the constructor after simplifying it a little.

Reply:
I noticed in the tests that `uri` can also be a regexp, that's why I was checking for the existence of `indexOf`.